### ISSUE TITLE
all-in-one command for crud generation

### ISSUE THREAD

**Issue by vesper8** [Laravel-Backpack/CRUD#1772]
Just thought I would share my custom command for adding a new model to the crud, adding the route and adding the sidebar item. It also makes sure not to add duplicate entries in the routes/sidebar

This is simply based on your commands in the getting started guide

I think it would make sense to include such a command built-in

```
php artisan backpack:crud:all tag
```

```
use Artisan;
use Illuminate\Console\Command;
use Illuminate\Foundation\Bus\DispatchesJobs;

class BackpackCrudAll extends Command
{
    use DispatchesJobs;

    protected $signature = 'backpack:crud:all {model}';

    protected $description = 'Creates a backpack model/request/controller/route/sidebar item';

    public function handle()
    {
        $arguments = $this->argument();
        $options = $this->option();

        $model = [
            'lowercase' => strtolower($arguments['model']),
            'ucfirst' => ucfirst($arguments['model']),
        ];

        // create a model, a request and a controller for the admin panel
        Artisan::call('backpack:crud', [
            'name' => $model['lowercase'],
        ]);

        // add a route to routes/backpack/custom.php
        $routeCode = "CRUD::resource('{$model['lowercase']}', '{$model['ucfirst']}CrudController');";
        $routesFileContent = file_get_contents(base_path('routes/backpack/custom.php'));
        if (!strpos($routesFileContent, $routeCode)) {
            Artisan::call('backpack:base:add-custom-route', [
                'code' => $routeCode,
            ]);
        } else {
            echo "Route already exists!\n";
        }

        // add a sidebar item
        $sidebarCode = "<li><a href='{{ backpack_url('{$model['lowercase']}') }}'><i class='fa fa-{$model['lowercase']}'></i> <span>{$model['ucfirst']}s</span></a></li>";
        $sidebarFileContent = file_get_contents(base_path('resources/views/vendor/backpack/base/inc/sidebar_content.blade.php'));
        if (!strpos($sidebarFileContent, $sidebarCode)) {
            Artisan::call('backpack:base:add-sidebar-content', [
                'code' => $sidebarCode,
            ]);
        } else {
            echo "Sidebar item already exists!\n";
        }
    }
}
```

**Comment by tabacitu** [comment]
Thank you for sharing your solution @vesper8 ! 

We're actually working on this problem _right now_. I'm working on a command-line wizard, to make this super-easy, using [this excellent package](https://github.com/webfactor/laravel-generators) that @OliverZiegler and @tswonke have built. You can already use their package, it works great, what I'm trying to do is just making it _even easier_ to use. Heavily recommend using their package - it'll come as part of Backpack itself very soon.

**Comment by vesper8** [comment]
good stuff! I probably will keep using my own for the added flexibility though.

Like for example I've now modified the script I posted above even further to not create the models and to link the generated crud controllers to my existing models, which happen to be in a different location than you would expect.. outside the app folder too

I also found a mighty useful package in the last few days that helps me use the artisan make commands for my purposes (files outside the app folder with my own namespace that doesn't start with App/

It's this one: https://github.com/diego-rlima/artisan-make-file-location

Glad you're working on improvements to the flow!
